hedera-node/src/main/java/com/hedera/services/ledger/AccountsCommitInterceptor.java

```diff
@@ -0,0 +1,82 @@
+package com.hedera.services.ledger;
+
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hederahashgraph.api.proto.java.AccountID;
+
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Singleton
+public class AccountsCommitInterceptor implements
+		CommitInterceptor<AccountID, MerkleAccount, AccountProperty> {
+
+	// The tracker this interceptor should use for previewing changes. The interceptor is NOT
+	// responsible for calling reset() on the tracker, as that will be done by the client code.
+	private final SideEffectsTracker sideEffectsTracker;
+
+	public AccountsCommitInterceptor(final SideEffectsTracker sideEffectsTracker) {
+		this.sideEffectsTracker = sideEffectsTracker;
+	}
+
+	/**
+	 * Accepts a pending change set, including creations and removals.
+	 *
+	 * @throws IllegalStateException if these changes are invalid
+	 */
+	@Override
+	public void preview(final List<MerkleLeafChanges<AccountID, MerkleAccount, AccountProperty>> changesToCommit) {
+		final List<Long> balances = new ArrayList<>();
```

Comment:
We don't need to construct a separate list here; a `long sum` variable that we update `sum += balanceChange` would be much more efficient.